### PR TITLE
Save in local storage

### PR DESCRIPTION
**Pull request by Timo-Thees**
Included in this feature:
- Projects are now saved in local storage
- added the option to overwrite a project if both projects have the same name
- added option to delete projects
- popup to reduce risk of accidental deletions

### CODE REVIEW

File: components/Editor.js [hunk 1]
```diff
@@ -1,35 +1,60 @@
...
+      {finalCount.closingMessage === true &&
+      nameTakenContent.taken === false ? (
+        <ClosingPopup
+          finalCount={finalCount}
+          setFinalCount={setFinalCount}
+          handleChangePage={handleChangePage}
+        />
+      ) : (
+        <></>
+      )}
```
Comment (stefankorder) [inline components/Editor.js:38]: You could write it also like this:

```suggestion
      {finalCount.closingMessage &&
      !nameTakenContent.taken && (
        <ClosingPopup
          finalCount={finalCount}
          setFinalCount={setFinalCount}
          handleChangePage={handleChangePage}
        />
      )}
```

File: components/Projects.js [hunk 2]
```diff
@@ -19,11 +24,25 @@ export default function Projects({
...
+      {deletePopup === true ? (
+        <DeleteConfirm
+          setDeletePopup={setDeletePopup}
+          idForDeletion={idForDeletion}
+          myProjects={myProjects}
+          setMyProjects={setMyProjects}
+        />
+      ) : (
+        <></>
+      )}
```
Comment (stefankorder) [inline components/Projects.js:45]: You could also write it like this:

```suggestion
      {deletePopup && (
        <DeleteConfirm
          setDeletePopup={setDeletePopup}
          idForDeletion={idForDeletion}
          myProjects={myProjects}
          setMyProjects={setMyProjects}
        />
      )}
```

**Review by stefankorder — CHANGES_REQUESTED** [review]
Good work! Take a look at my comments. Maybe there are a few other places you could implement my suggestions.

File: components/Projects.js (hunk 2 above)
Comment (Timo-Thees) [reply components/Projects.js]: Implementing the code changes produces a prettier error

**Comment by Timo-Thees** [comment]
Thank you for your review! I am aware that there must be a more elegant way to write the annotated lines in the code. However, implementing the requested changes resulted in a failure of the first test. Therefore I reverted back to the previous code

File: components/Editor.js [hunk 3]
```diff
@@ -1,35 +1,57 @@
...
     const text = event.target.text.value;
     const id = editorContent.id;
-    handleChangePage("my projects");
-    saveProjects(title, text, id);
+    const finalWordCount = wordcount - editorContent.wordcount;
+    setSessionClose({wordcount: finalWordCount, showClosingMessage: true});
+    saveProjects(title, text, id, wordcount);
   };
   return (
     <div>
```
Comment (stefankorder) [inline components/Editor.js:28]: Maybe you could use a Fragment `<>` insgtead a `<div>` at this point.

File: pages/index.js [hunk 4]
```diff
@@ -51,9 +69,11 @@ export default function Home() {
         <NameTaken
```
Comment (stefankorder) [inline pages/index.js:69]: You could use a logical "and" `&&` instead a "ternary" `? :` in such cases where the falsy value is just a Fragment or an empty string.

**Review by stefankorder — APPROVED** [review]
Good work! Just 2 hints.

**Pull request merged**